akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/IntegrationDocTest.java

```diff
@@ -0,0 +1,567 @@
+      
+      {
+        //#sometimes-slow-mapAsync
+        final MessageDispatcher blockingEc = system.dispatchers().lookup("blocking-dispatcher");
+        final SometimesSlowService service = new SometimesSlowService(blockingEc);
+
+        final ActorFlowMaterializer mat = ActorFlowMaterializer.create(
+          ActorFlowMaterializerSettings.create(system).withInputBuffer(4, 4), system);
+
+        Source.from(Arrays.asList("a", "B", "C", "D", "e", "F", "g", "H", "i", "J"))
+          .map(elem -> { println("before: " + elem); return elem; })
+          .mapAsync(service::convert)
+          .runForeach(elem -> println("after: " + elem), mat);
+        //#sometimes-slow-mapAsync
+
+        probe.expectMsg("after: A");
+        probe.expectMsg("after: B");
+        probe.expectMsg("after: C");
+        probe.expectMsg("after: D");
+        probe.expectMsg("after: E");
+        probe.expectMsg("after: F");
+        probe.expectMsg("after: G");
+        probe.expectMsg("after: H");
+        probe.expectMsg("after: I");
+        probe.expectMsg("after: J");
+      }
+    };
+  }
+  
+  @Test
+  public void illustrateOrderingAndParallelismOfMapAsyncUnordered() throws Exception {
+    new JavaTestKit(system) {
+      final TestProbe probe = new TestProbe(system);
+      final EmailServer emailServer = new EmailServer(probe.ref());
+
+      void println(String s) {
+        if (s.startsWith("after:"))
+          probe.ref().tell(s, ActorRef.noSender());
+      }
+      
+      {
+        //#sometimes-slow-mapAsyncUnordered
+        final MessageDispatcher blockingEc = system.dispatchers().lookup("blocking-dispatcher");
+        final SometimesSlowService service = new SometimesSlowService(blockingEc);
+
+        final ActorFlowMaterializer mat = ActorFlowMaterializer.create(
+          ActorFlowMaterializerSettings.create(system).withInputBuffer(4, 4), system);
+
+        Source.from(Arrays.asList("a", "B", "C", "D", "e", "F", "g", "H", "i", "J"))
+          .map(elem -> { println("before: " + elem); return elem; })
+          .mapAsyncUnordered(service::convert)
+          .runForeach(elem -> println("after: " + elem), mat);
+        //#sometimes-slow-mapAsyncUnordered
+
+        Seq<Object> got = probe.receiveN(10);
+        Set<Object> set = new HashSet<>(got.size());
+        for (int i = 0; i < got.size(); i++) {
+          set.add(got.apply(i));
+        }
+        assertTrue(set.contains("after: A"));
+        assertTrue(set.contains("after: B"));
+        assertTrue(set.contains("after: C"));
+        assertTrue(set.contains("after: D"));
+        assertTrue(set.contains("after: E"));
+        assertTrue(set.contains("after: F"));
+        assertTrue(set.contains("after: G"));
+        assertTrue(set.contains("after: H"));
+        assertTrue(set.contains("after: I"));
+        assertTrue(set.contains("after: J"));
+      }
+    };
+  }
+  
+  
+  
+  @SuppressWarnings("UnusedParameters")
+  static <T> void println(T s) { /* ignore */ }
```

Comment:
yes, I was also thinking about how we should do that. The problem is that the user can't copy the code and just run it, since this doesn't exist. I don't know.